Avoid to iterate too fast on server looping and protect CPU usage

eusonlito opened the pull request:
Maybe related with https://github.com/arthurkushman/php-wss/issues/73

I have also a high CPU usage once the remote clients are connected to websocket server.

Adding a 0.001 seconds delay on every server loop iteration can reduce about a 90% the CPU usage without appreciable loss of performance.

Also can be changed to 0.01 seconds delay to reduce about a 99% de CPU usage without appreciable performance loss.

arthurkushman commented on src/WebSocketServer.php:
```diff
@@ -52,6 +52,10 @@ class WebSocketServer extends WssMain implements WebSocketServerContract
     /**
+     * @var int
+     */
+    private int $loopingDelay = 1000;
```
Does this delay time for 1ms solves the problem with CPU high consuming in practice? Was there any metrics? Is it 20%, 40% etc less f.e.?

eusonlito replied:
I'm using my own fork. With a 1ms delay the server CPU goes from 100% or more to about 6%. And using 2ms, the CPU is going about 2%.

eusonlito replied:
We can ask for milliseconds instead of microseconds of a second, and then set the usleep value x1000.

arthurkushman replied:
> We can ask for milliseconds instead of microseconds of a second, and then set the usleep value x1000.

Sure it would be reasonable

arthurkushman commented on src/WebSocketServer.php:
```diff
@@ -74,6 +78,18 @@ public function __construct(
+     *
+     * @return self
+     */
+    public function loopingDelay(int $loopingDelay): self
```
It would be better to set up some thresholds min/max, because if user will set up say 1 it will be too low to make things better for CPU and rather consume more for nothing and vice versa 10 000 000 let it sleep for 10 seconds in the loop just for nothing. Also, is it a good idea to let users choose this amount of delay? Probably constant value will be more of a fit here?

eusonlito replied:
Yes, we can add some fixed thresholds :)

eusonlito replied:
And we can ask for milliseconds instead of millionths of a second.

eusonlito commented:
* Moved the `loopingDelay` method to `ServerConfig` as `setLoopingDelay`.
* Added `loopingDelay` value control using `loopingDelayRange`
* Added config option to README.
